src/ui/page_version.c

```diff
@@ -128,35 +128,35 @@ int generate_current_version(sys_version_t *sys_ver) {
     if (!fp) {
         return -1;
     }
-    fscanf(fp, "%d.%d.%d-%s",
+    fscanf(fp, "%hhd.%hhd.%hhd-%s",
```

Comment:
seeing we are after correctness, these should be `%hhu`, as it's a unsigned byte

Reply:
Agreed.  Will update.